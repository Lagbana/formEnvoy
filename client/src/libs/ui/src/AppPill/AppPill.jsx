/* Pill shape (icon + text)
=============================== */

import React from 'react';
import { StyleSheet, View, Text, ViewStyle } from 'react-native';
import { Typography, Colors } from 'libs/ui/src/WebButton/node_modules/libs/ui/src/WebLoading/node_modules/colibri/styles';
import { AppIcon, GLYPHS } from '../AppIcon';

export interface AppPillProps {
  icon?: GLYPHS;
  label?: string;
  color?: string;
  style?: ViewStyle;
}

export const AppPill: React.FC<AppPillProps> = ({
  icon,
  label = '',
  color = 'blue',
  style,
}) => {
  // styles

  const styles = StyleSheet.create({
    container: {
      flexDirection: 'row',
      alignItems: 'center',
      marginLeft: 'auto',
      marginRight: 'auto',
      paddingVertical: 7,
      paddingHorizontal: 15,
      backgroundColor: Colors[color],
      borderRadius: 99,
    },
    icon: {
      color: '#000',
      opacity: 0.45,
      marginTop: 2,
      marginRight: 6,
    },
    label: {
      ...Typography.labelSmall,
      color: Colors.white,
    },
  });

	const iconDisplay = (icon) ? <AppIcon name={icon} size={18} style={styles.icon} /> : null;

  return (
    <View style={[styles.container, style]}>
			{iconDisplay}
      <Text style={styles.label}>{label}</Text>
    </View>
  );
};
