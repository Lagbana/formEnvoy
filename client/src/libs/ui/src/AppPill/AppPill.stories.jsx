import React from 'react'
import { storiesOf } from '@storybook/react-native'
import { AppPill } from './AppPill'

storiesOf('AppPill', module)
  .add('Smiling face', () => (
    <AppPill icon={'smiling-face'} label='Smiling face pill' color={'blue'} />
  ))
  .add('Laughing face', () => (
    <AppPill icon={'laughing-face'} label='Laughing face pill' color={'blue'} />
  ))
  .add('Neutral face', () => (
    <AppPill icon={'neutral-face'} label='Neutral face pill' color={'blue'} />
  ))
