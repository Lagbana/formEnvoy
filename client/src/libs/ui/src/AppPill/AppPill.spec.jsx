import React from 'react'
import { shallow, configure } from 'enzyme'
import toJson from 'enzyme-to-json'
import Adapter from 'enzyme-adapter-react-16'
import { AppPill } from './AppPill'

configure({ adapter: new Adapter() })

test('<AppPill /> renders correctly', () => {
  const component = shallow(<AppPill />)
  component.unmount()
  expect(toJson(component)).toMatchSnapshot()
})
