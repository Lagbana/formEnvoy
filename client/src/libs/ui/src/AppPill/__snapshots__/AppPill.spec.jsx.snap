// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppPill /> renders correctly 1`] = `""`;
